Editor/Window/UserProfileSelection.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Linq;
+using System.Threading;
+using Editor.CoreAPI;
+using UnityEngine.UIElements;
+
+namespace AmazonGameLift.Editor
+{
+    internal class UserProfileSelection
+    {
+        private BootstrapSettings _bootstrapSettings;
+        private CancellationTokenSource _refreshBucketsCancellation;
+        private readonly VisualElement _container;
+        private readonly AwsUserProfilesPage _awsUserProfilesPage;
+        private readonly StateManager _stateManager;
+        
+        public UserProfileSelection(VisualElement container, StateManager stateManager, AwsUserProfilesPage profilesPage)
+        {
+            _container = container;
+            _stateManager = stateManager;
+            _awsUserProfilesPage = profilesPage;
+            _bootstrapSettings = profilesPage.BootstrapSettings;
+        }
+
+        private void BucketSelection()
+        {
+            _refreshBucketsCancellation = new CancellationTokenSource();
+            _ = _bootstrapSettings.RefreshExistingBuckets(_refreshBucketsCancellation.Token);
+            if (_bootstrapSettings.BucketName != null)
+            {
+                BucketSelection(_bootstrapSettings.BucketName);
+                _bootstrapSettings.SaveSelectedBucket();
+            }
+        }
+
+        public void BucketSelection(string selectedBucket)
+        {
+            _bootstrapSettings.SelectBucket(selectedBucket);
+            _bootstrapSettings.SaveSelectedBucket();
+            _container.Q<Label>("S3BucketNameLabel").text = selectedBucket;
+        }
+
+        public void AccountSelection(bool isSetup)
```

Comment:
Please rename to `SelectAccount`for reasons outlined above.

Reply:
Or rather `SelectProfile`since this is meant to toggle profiles

Reply:
Renamed

Reply:
UserProfileSelection is no longer used, removed;